pkg/networkservice/common/begin/merge.go

```diff
@@ -36,7 +37,8 @@ func mergeConnection(returnedConnection, requestedConnection, connection *networ
 func mergeConnectionContext(returnedConnectionContext, requestedConnectionContext, connectioncontext *networkservice.ConnectionContext) *networkservice.ConnectionContext {
 	rv := proto.Clone(connectioncontext).(*networkservice.ConnectionContext)
 	if !proto.Equal(returnedConnectionContext, requestedConnectionContext) {
-		// TODO: IPContext, DNSContext, EthernetContext, do we need to do MTU?
+		// TODO: DNSContext, EthernetContext, do we need to do MTU?
+		rv.IpContext = requestedConnectionContext.IpContext
```

Comment:
Hmm, I think we also need to do it for other context as you pointed in the comment.


@edwarnicke Thoughts?

Reply:
@LionelJouin Apologies for this taking so long to get back to you on. The net net is that what you did here was fine.  I also realized why it took me so long to sort it out, and produced a better documented version here in #1237 that handles the merge more or less correctly (which you were doing) and also in a way that's hopefully clearer.  I also provided testing.  Could you have a look?

Reply:
no problem, I will try it today, thank you